daemon/copy.go

```diff
@@ -1,7 +1,34 @@
 package daemon
 
-import "io"
+import (
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
 
+	"github.com/docker/docker/pkg/archive"
+	"github.com/docker/docker/pkg/chrootarchive"
+	"github.com/docker/docker/pkg/ioutils"
+)
+
+// ErrExtractPointNotDirectory is used to convey that the operation to extract
+// a tar archive to a directory in a container has failed because the specified
+// path does not refer to a directory.
+var ErrExtractPointNotDirectory = errors.New("extraction point is not a directory")
+
+// ContainerResourceStat describes a file or directory resource inside a
+// container.
+type ContainerResourceStat struct {
```

Comment:
see comment above about removing this here and having one struct in api/types/types.go that both the cli and server can use.